src/MetaMorpho.sol

```diff
@@ -367,12 +371,23 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
                 if (config[id].cap != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroCap(id);
 
-                if (MORPHO.supplyShares(id, address(this)) != 0) {
+                uint256 supplyShares = MORPHO.supplyShares(id, address(this));
+
+                if (supplyShares != 0) {
                     if (config[id].removableAt == 0) revert ErrorsLib.InvalidMarketRemovalNonZeroSupply(id);
 
                     if (block.timestamp < config[id].removableAt) {
                         revert ErrorsLib.InvalidMarketRemovalTimelockNotElapsed(id);
                     }
+
+                    bytes32[] memory slot = new bytes32[](1);
+                    slot[0] = id.marketTotalSupplyAssetsAndSharesSlot();
```

Comment:
Funny slot calculation, i didn't expect that
It's fine to me, but I think i prefer `MorphoStorageLib.marketTotalSupplyAssetsAndSharesSlot(id)`